pyfakefs/fake_filesystem.py

```diff
@@ -1726,58 +1782,61 @@ def _starts_with_drive_letter(self, file_path):
                     return False
         return False
 
-    def _starts_with_root_path(self, file_path):
+    def _starts_with_root_path(self, file_path: AnyStr) -> bool:
         root_name = matching_string(file_path, self.root.name)
         file_path = self._normalize_path_sep(file_path)
         return (file_path.startswith(root_name) or
                 not self.is_case_sensitive and file_path.lower().startswith(
                     root_name.lower()) or
                 self._starts_with_drive_letter(file_path))
 
-    def _is_root_path(self, file_path):
+    def _is_root_path(self, file_path: AnyStr) -> bool:
         root_name = matching_string(file_path, self.root.name)
         return (file_path == root_name or not self.is_case_sensitive and
                 file_path.lower() == root_name.lower() or
                 2 <= len(file_path) <= 3 and
                 self._starts_with_drive_letter(file_path))
 
-    def ends_with_path_separator(self, file_path):
+    def ends_with_path_separator(self, path: Union[int, AnyPath]) -> bool:
         """Return True if ``file_path`` ends with a valid path separator."""
-        if is_int_type(file_path):
+        if is_int_type(path):
             return False
-        file_path = make_string_path(file_path)
-        return (file_path and
-                file_path not in (self.path_separator,
-                                  self.alternative_path_separator) and
-                (file_path.endswith(self._path_separator(file_path)) or
-                 self.alternative_path_separator is not None and
-                 file_path.endswith(
-                     self._alternative_path_separator(file_path))))
-
-    def is_filepath_ending_with_separator(self, path):
+        file_path = os.fspath(path)  # type: ignore[arg-type]
```

Comment:
Argument 1 to "fspath" has incompatible type "Union[int, Union[str, bytes, _PathLike[Any]]]"; expected "str"

Reply:
You could probably fix this with an `isinstance` guard beforehand

Reply:
Fixed meanwhile.